### ISSUE TITLE
Compilation failure on ARMv7

### ISSUE THREAD

**Issue by benthetechguy** [PolyMC/PolyMC#1116]
### Operating System

Linux

### Version of PolyMC

PolyMC 1.4.1

### Version of Qt

Qt 6.3.1

### Description of bug

The build fails while on ARMv7 (armhf) when linking `../Library` with the error `multiple definition of 'typeinfo name for nbt::tag_primitive<signed char>'`. A build log is attached [here](https://github.com/PolyMC/PolyMC/files/9501783/log.txt).

### Steps to reproduce

1. Install your favorite ARMv7 Linux distribution (Debian and Arch Linux ARM tested)
2. Install build dependencies
3. Compile (with or without Qt6 flag, still occurs)
4. Notice error 95% of the way through and compilation fails

### Suspected cause

I searched launcher/World.cpp, which is the file it complains about, and did not find a mention of `nbt::tag_primitive`, so no idea where to start here - maybe libnbtplusplus? If it ends up being a libnbt++ issue, this can be closed and a new one opened there.

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by DioEgizio** [comment]
Inherited from multimc, see https://github.com/MultiMC/Launcher/issues/2895

TL;DR GCC issue, use clang

**Comment by Scrumplex** [comment]
> Inherited from multimc, see [MultiMC/Launcher#2895](https://github.com/MultiMC/Launcher/issues/2895)
> 
> TL;DR GCC issue, use clang

But here we have a very up-to-date GCC. If you look at the build log there are multiple warnings while compiling so there might be a deeper issue

**Comment by DioEgizio** [comment]
up to date GCC doesn't mean the issue is fixed; the void package, for example, still blacklists arm 32 bit because of that

**Comment by benthetechguy** [comment]
Just tested building with Clang, and it works. This is still an issue, however; I will test GCC on more architectures to see if this is only a problem with ARMv7.

**Comment by benthetechguy** [comment]
I have tried every architecture supported by Debian, and all except armhf and armel (same issue) built successfully.

**Comment by theofficialgman** [comment]
This is likely a longstanding GCC bug. I doubt it will be fixed unless someone knows who develops the armv7 compiler for GCC and knows what buttons to push/people to notify.

For the people it affects, you should probably start reading here: https://gcc.gnu.org/bugs/
